x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/fields/field_map.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FieldMap, alertFieldMap } from '@kbn/alerts-as-data-utils';
+import {
+  ALERT_ATTACK_DISCOVERY_ALERTS_CONTEXT_COUNT,
+  ALERT_ATTACK_DISCOVERY_ALERT_IDS,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_ACTION_TYPE_ID,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_CONNECTOR_ID,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_DEFAULT_SYSTEM_PROMPT_ID,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_MODEL,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_PROVIDER,
+  ALERT_ATTACK_DISCOVERY_DETAILS_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_ENTITY_SUMMARY_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_MITRE_ATTACK_TACTICS,
+  ALERT_ATTACK_DISCOVERY_REPLACEMENTS,
+  ALERT_ATTACK_DISCOVERY_SUMMARY_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_TITLE,
+  ALERT_ATTACK_DISCOVERY_USERS,
+  ALERT_ATTACK_DISCOVERY_USER_ID,
+  ALERT_ATTACK_DISCOVERY_USER_NAME,
+} from './field_names';
+
+export const attackDiscoveryAlertFieldMap: FieldMap = {
+  /**
+   * Default alert-as-data fields
+   */
+  ...alertFieldMap,
+
+  /**
+   * Attack discovery fields
+   */
+  [ALERT_ATTACK_DISCOVERY_USERS]: {
+    type: 'nested',
+    array: true,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_USER_ID]: {
+    type: 'keyword',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_USER_NAME]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_DETAILS_MARKDOWN]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_TITLE]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_ENTITY_SUMMARY_MARKDOWN]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_SUMMARY_MARKDOWN]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_MITRE_ATTACK_TACTICS]: {
+    type: 'keyword',
+    array: true,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_ALERT_IDS]: {
+    type: 'keyword',
+    array: true,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_REPLACEMENTS]: {
+    type: 'object',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG]: {
+    type: 'object',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_CONNECTOR_ID]: {
+    type: 'keyword',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_ACTION_TYPE_ID]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_DEFAULT_SYSTEM_PROMPT_ID]: {
```

Comment:
Consider removing `ALERT_ATTACK_DISCOVERY_API_CONFIG_DEFAULT_SYSTEM_PROMPT_ID`, because Attack discovery does not have a user-configurable system prompt.

Reply:
I was thinking that we might want to store connector ID only. Other fields can be updated by the user at any point and will not make sense anymore? Should not we try to use `joint` fetch and enrich alerts with connector information in the UI? We do something similar with the alert assignments where we pre-fetch user profiles.

Reply:
> I was thinking that we might want to store connector ID only. Other fields can be updated by the user at any point and will not make sense anymore? Should not we try to use `joint` fetch and enrich alerts with connector information in the UI? We do something similar with the alert assignments where we pre-fetch user profiles.

- For the specific case of the `ALERT_ATTACK_DISCOVERY_API_CONFIG_DEFAULT_SYSTEM_PROMPT_ID` field, it doesn't appear that this field would be necessary for Attack discovery. 

- For the general case of joining data in the UI (via multiple API calls) vs denormalizing, I feel like [this comment](https://github.com/elastic/kibana/pull/215184/files#r2004353809) above addresses some of the challenges associated with doing "joins" with connectors in the UI.